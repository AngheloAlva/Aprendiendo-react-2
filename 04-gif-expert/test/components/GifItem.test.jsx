import { render, screen } from '@testing-library/react'
import GifItem from '../../src/components/GifItem'

/* eslint-disable no-undef */
describe('Test para el componente GifItem', () => {
  const title = 'Batman'
  const url = 'https://localhost/algo.jpg'

  test('Debe haber un match con el snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />)
    expect(container).toMatchSnapshot()
  })

  test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
    render(<GifItem title={title} url={url} />)
    const { src, alt } = screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(title)
  })

  test('Debe estar el titulo en el documento', () => {
    render(<GifItem title={title} url={url} />)
    expect(screen.getByText(title)).toBeTruthy()
  })
})
