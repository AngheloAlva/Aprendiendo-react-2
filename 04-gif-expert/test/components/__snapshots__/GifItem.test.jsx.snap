// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test para el componente GifItem Debe haber un match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Batman"
      src="https://localhost/algo.jpg"
    />
    <p>
      Batman
    </p>
  </div>
</div>
`;
